zcash_client_sqlite/src/wallet.rs

```diff
@@ -1208,7 +1208,16 @@ pub(crate) fn truncate_to_height<P: consensus::Parameters>(
             params: params.clone(),
         };
         wdb.with_sapling_tree_mut(|tree| {
-            tree.truncate_removing_checkpoint(&block_height).map(|_| ())
+            let truncate_succeeded = tree.truncate_removing_checkpoint(&block_height)?;
```

Comment:
The odd `Result<bool, _>` return type from `ShardTree::truncate_removing_checkpoint` probably contributed to this bug.

https://github.com/zcash/incrementalmerkletree/blob/2ce18583d482d3868a2218657882c521e08f9c6d/shardtree/src/lib.rs#L562-L570